src/librustc_trans/save/mod.rs

```diff
@@ -652,6 +665,37 @@ impl<'l, 'tcx: 'l> SaveContext<'l, 'tcx> {
         })
     }
 
+    pub fn get_macro_use_data(&self, span: Span, id: NodeId) -> Option<MacroUseData> {
+        // The macro use has been expanded out of the AST at this point,
+        // but we can recreate it using the callsite and callee spans in
+        // the expansion trace.
+        // Note we take care to use the source callsite/callee, to handle
+        // nested expansions.
+        if !generated_code(span) {
+            return None;
+        }
+        let callsite = self.tcx.sess.codemap().source_callsite(span);
+        let callee = self.tcx.sess.codemap().source_callee(span);
+        if let None = callee {
```

Comment:
For this and the below `if let`, use [option_try](https://dxr.mozilla.org/rust/source/src/librustc/util/common.rs#73-75)